// igniteui-react-core/esm2015/lib/GeographicTileSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { GeographicShapeSeriesBaseDescription } from "./GeographicShapeSeriesBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let GeographicTileSeriesDescription = /*@__PURE__*/ (() => {
    class GeographicTileSeriesDescription extends GeographicShapeSeriesBaseDescription {
        constructor() {
            super();
            this.dc = null;
            this.de = null;
        }
        get_type() {
            return "GeographicTileSeries";
        }
        get tileImagery() {
            return this.dc;
        }
        set tileImagery(a) {
            this.dc = a;
            this.e("TileImagery");
        }
        get imageTilesReadyRef() {
            return this.de;
        }
        set imageTilesReadyRef(a) {
            this.de = a;
            this.e("ImageTilesReadyRef");
        }
    }
    GeographicTileSeriesDescription.$t = /*@__PURE__*/ markType(GeographicTileSeriesDescription, 'GeographicTileSeriesDescription', GeographicShapeSeriesBaseDescription.$);
    return GeographicTileSeriesDescription;
})();
