// igniteui-react-core/esm2015/lib/ImageCellInfoDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CellInfoDescription } from "./CellInfoDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let ImageCellInfoDescription = /*@__PURE__*/ (() => {
    class ImageCellInfoDescription extends CellInfoDescription {
        constructor() {
            super();
            this.fd = null;
            this.ff = null;
            this.fe = null;
        }
        get_type() {
            return "ImageCellInfo";
        }
        get imagePath() {
            return this.fd;
        }
        set imagePath(a) {
            this.fd = a;
            this.e("ImagePath");
        }
        get imageStretchOption() {
            return this.ff;
        }
        set imageStretchOption(a) {
            this.ff = a;
            this.e("ImageStretchOption");
        }
        get imageResourceType() {
            return this.fe;
        }
        set imageResourceType(a) {
            this.fe = a;
            this.e("ImageResourceType");
        }
    }
    ImageCellInfoDescription.$t = /*@__PURE__*/ markType(ImageCellInfoDescription, 'ImageCellInfoDescription', CellInfoDescription.$);
    ImageCellInfoDescription.__marshalByValue1 = true;
    ImageCellInfoDescription.__marshalByValueAlias1 = "ImageCellModel";
    return ImageCellInfoDescription;
})();
