// igniteui-react-core/esm2015/lib/PercentagePriceOscillatorIndicatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicatorDescription } from "./StrategyBasedIndicatorDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let PercentagePriceOscillatorIndicatorDescription = /*@__PURE__*/ (() => {
    class PercentagePriceOscillatorIndicatorDescription extends StrategyBasedIndicatorDescription {
        constructor() {
            super();
            this.dw = 0;
            this.dv = 0;
        }
        get_type() {
            return "PercentagePriceOscillatorIndicator";
        }
        get shortPeriod() {
            return this.dw;
        }
        set shortPeriod(a) {
            this.dw = a;
            this.e("ShortPeriod");
        }
        get longPeriod() {
            return this.dv;
        }
        set longPeriod(a) {
            this.dv = a;
            this.e("LongPeriod");
        }
    }
    PercentagePriceOscillatorIndicatorDescription.$t = /*@__PURE__*/ markType(PercentagePriceOscillatorIndicatorDescription, 'PercentagePriceOscillatorIndicatorDescription', StrategyBasedIndicatorDescription.$);
    return PercentagePriceOscillatorIndicatorDescription;
})();
